test/socket/SocketBatcher.t.sol

```diff
@@ -0,0 +1,203 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.19;
+
+import "../Setup.t.sol";
+import "../../contracts/socket/SocketBatcher.sol";
+import "../../contracts/examples/Counter.sol";
+
+contract SocketBatcherTest is Setup {
+    Counter srcCounter__;
+    Counter dstCounter__;
+    SocketBatcher batcher__;
+
+    uint256 addAmount = 100;
+    uint256 subAmount = 40;
+    bool isFast = true;
+    uint256 index = isFast ? 0 : 1;
+
+    bytes32[] roots;
+
+    event ExecutionSuccess(bytes32 msgId);
+    event ExecutionFailed(bytes32 msgId, string result);
+    event ExecutionFailedBytes(bytes32 msgId, bytes result);
+    event PacketVerifiedAndSealed(
+        address indexed transmitter,
+        bytes32 indexed packetId,
+        bytes32 root,
+        bytes signature
+    );
+
+    function setUp() external {
+        uint256[] memory transmitterPrivateKeys = new uint256[](1);
+        transmitterPrivateKeys[0] = _transmitterPrivateKey;
+
+        _dualChainSetup(transmitterPrivateKeys);
+        _deployPlugContracts();
+
+        _configPlugContracts(index);
+        batcher__ = new SocketBatcher(address(this));
+    }
```

Comment:
You need to learn mocking and unit testing. Using expectCall and mockCall would have been very easy and ideal for this contract. Full integration test seems overkill.